fastify.js

```diff
@@ -245,6 +245,37 @@ function build (options) {
     }
   }
 
+  function listenPromise (port, address, backlog) {
+    address = address || '127.0.0.1'
+
+    if (listening) {
+      return Promise.reject(new Error('Fastify is already listening'))
+    }
+
+    return fastify.ready().then(() => {
+      listening = true
+
+      var errEventHandler
+      var errEvent = new Promise((resolve, reject) => {
+        errEventHandler = (err) => reject(err)
+        server.once('error', errEventHandler)
+      })
+      var listen = new Promise((resolve, reject) => {
+        server.listen(port, address, backlog, (err) => {
+          if (err) reject(err)
+          else resolve()
+        })
+      })
+
+      return Promise.race([
+        errEvent, // e.g invalid port range error is always emitted before the server listening
+        listen
+      ])
+        .then(() => server.removeListener('error', errEventHandler))
```

Comment:
can you move this removeListener line inside the listen callback?

Reply:
yes, much better.